File: python/equalize.py
# program to count minimum number of +1 or -1 operations required to make all the elements of the array equal
# arr = [1, 2, 7, 4, 10]
"""
sum = 24
len = 5
mean = 24 // 5
mean = 4
[1, 2, 7, 4, 10] -mean
return sum([3, 2, 3, 0, 6])
"""


def equalize(arr):
    mean = sum(arr) // len(arr)
    return sum([abs(mean - i) for i in arr])


print('Min steps required  : ', equalize(list(map(int, input('Enter an array : ').split()))))

File: c++/equalize.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    vector<int> arr;
    cout<<"Enter number of elements : ";
    int n; 
    cin>>n;
    cout<<"Enter array elements separated by space : ";
    for (int i=0; i<n; i++)
        {
            int temp;
            cin>>temp;
            arr.push_back(temp);
        }

    int count = 0;
    int sum = 0;
    int mean = accumulate(arr.begin(), arr.end(), sum) / n;
    for (int i : arr)
        count += abs(i - mean);
    cout<<"Min steps required  : "<<count<<endl;
    return 0;
}
